src/main/java/carpet_client/utils/CarpetRules.java

```diff
@@ -0,0 +1,177 @@
+import net.minecraft.util.PacketByteBuf;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class CarpetRules
+{
+    public static PacketByteBuf data;
+    private static final Map<String, CarpetSettingEntry> rules = new HashMap<>();
+    
+    public static CarpetSettingEntry getRule(String rule)
+    {
+        return rules.get(rule);
+    }
+    
+    public static boolean hasRule(String rule)
+    {
+        return rules.containsKey(rule);
+    }
+    
+    public static void setAllData(PacketByteBuf buf)
+    {
+        CarpetRules.data = buf;
+        split();
+        editClientRules();
+    }
+    
+    private static void editClientRules()
+    {
+    
+    }
+    
+    private static void split()
+    {
+        CompoundTag compound = data.readCompoundTag();
+        if (compound == null) return;
+        String carpetServerVersion = compound.getString("CarpetVersion");
+    
+        ListTag rulesList = compound.getList("rules", 10);
+        for (Tag tag : rulesList)
+        {
+            CompoundTag ruleNBT = (CompoundTag) tag;
+            String rule = ruleNBT.getString("rule");
+            String value = ruleNBT.getString("value");
+            String defaultOption = ruleNBT.getString("default");
+            String type = ruleNBT.getString("type");
+            String description = ruleNBT.getString("description");
+            if (hasRule(rule))
+                getRule(rule).update(value, null, defaultOption);
+            else
+                rules.put(rule, new CarpetSettingEntry(rule, value, null, defaultOption, type, description));
+        }
+        ConfigScreen.setCarpetServerVersion(carpetServerVersion);
+        Reference.isCarpetServer = true;
+    }
+    
+    public static void ruleData(PacketByteBuf data)
+    {
+        String rule = data.readString();
+        String newValue = data.readString();
+        getRule(rule).changeRule(newValue);
+    }
+    
+    public static void ruleChange(String rule, String newValue, MinecraftClient client)
+    {
+        PacketByteBuf data = new PacketByteBuf(Unpooled.buffer());
+        data.writeVarInt(Reference.RULE_REQUEST);
+        data.writeString(rule);
+        data.writeString(newValue);
+        ClientMessageHandler.sendPacket(data, client);
+    }
+    
+    public static void requestUpdate(MinecraftClient client)
+    {
+        PacketByteBuf data = new PacketByteBuf(Unpooled.buffer());
+        data.writeVarInt(Reference.ALL_GUI_INFO);
+        ClientMessageHandler.sendPacket(data, client);
+    }
+    
+    public static ArrayList<CarpetSettingEntry> getAllRules()
+    {
+        ArrayList<CarpetSettingEntry> res = new ArrayList<>();
+        for (String rule : rules.keySet().stream().sorted().collect(Collectors.toList()))
+        {
+            res.add(rules.get(rule));
+        }
+        return res;
+    }
+    
+    public static class CarpetSettingEntry
```

Comment:
This looks like some really old stuff from the old settings class. Probably will be using ParsedRule, or a subclass of it like ClientParsedRule at some point